### ISSUE TITLE
Layout/ClassStructure ignores `private_class_method`

### ISSUE THREAD

**Issue by marcandre** [rubocop/rubocop#9570]
```ruby
      expect_offense(<<~RUBY)
        class A
          private_class_method def self.foo
          end

          public_class_method def self.bar
          ^^^^^^^^^^^^^^ `public_class_methods` is supposed to appear before `private_class_methods`.
          end
        end
      RUBY

      expect_correction(<<~RUBY)
        class A
          public_class_method def self.bar
          end

          private_class_method def self.foo
          end
        end
      RUBY
```

Fails, no offense raised.

**Comment by bbatsov** [comment]
Reopened — still reproduces on master (1.88). `Layout/ClassStructure` ignores `private_class_method`/`public_class_method` ordering: a `public_class_method def self.bar` placed after `private_class_method def self.foo` raises no offense, while the equivalent instance-level `public`/`private` case is flagged.
